Utilities/File.cpp

```diff
@@ -1831,7 +1831,7 @@ bool fs::remove_all(const std::string& path, bool remove_root, bool is_no_dir_ok
 	return true;
 }
 
-u64 fs::get_dir_size(const std::string& path, u64 rounding_alignment)
+u64 fs::get_dir_size(const std::string& path, u64 rounding_alignment, atomic_t<bool>* cancel_flag)
```

Comment:
why not pass a shared_ptr?

Reply:
I don't see a reason to force it here.

Reply:
One reason would be to prevent regressions or edge cases that lead to the destruction of the source pointer before the threads are finished.

Reply:
The lambda captures the shared pointer by value not reference, it is destructed when the lambda itself is destructed (the code inside would be unreachable). As long as any shared pointer instance references the address, the address would not be deallocated.